sdk/python/foundation-models/system/content-safety/deploy_with_content_safety.ipynb

```diff
@@ -0,0 +1,803 @@
+    "    parameters=RoleAssignmentCreateParameters(\n",
+    "        role_definition_id=role_def.id,\n",
+    "        principal_id=uai_principal_id,\n",
+    "        principal_type=\"ServicePrincipal\",\n",
+    "    ),\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### 8.4 Assign AcrPull at the workspace container registry scope\n",
+    "Since we will create the safety enabled llama endpoint with User Assigned Identity, the user's managed identity must have Storage blob data reader permission on the storage account for the workspace, and AcrPull permission on the Azure Container Registry (ACR) for the workspace. Make sure your User Assigned Identity has the right permission."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "workspace = ml_client.workspaces.get(workspace_name)\n",
+    "container_registry = workspace.container_registry\n",
+    "\n",
+    "role_name = \"AcrPull\"\n",
+    "acr_scope = container_registry\n",
+    "\n",
+    "role_defs = role_definition_client.role_definitions.list(scope=acr_scope)\n",
+    "role_def = next((r for r in role_defs if r.role_name == role_name))\n",
+    "\n",
+    "role_assignment_client.role_assignments.create(\n",
+    "    scope=acr_scope,\n",
+    "    role_assignment_name=str(uuid.uuid4()),\n",
+    "    parameters=RoleAssignmentCreateParameters(\n",
+    "        role_definition_id=role_def.id,\n",
+    "        principal_id=uai_principal_id,\n",
+    "        principal_type=\"ServicePrincipal\",\n",
+    "    ),\n",
+    ")\n",
+    "print(\"Role assignment for AcrPull at the workspace container registry completed.\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "role_name = \"Storage Blob Data Reader\"\n",
+    "blob_scope = workspace.storage_account\n",
+    "\n",
+    "role_defs = role_definition_client.role_definitions.list(scope=blob_scope)\n",
+    "role_def = next((r for r in role_defs if r.role_name == role_name))\n",
+    "\n",
+    "role_assignment_client.role_assignments.create(\n",
+    "    scope=blob_scope,\n",
+    "    role_assignment_name=str(uuid.uuid4()),\n",
+    "    parameters=RoleAssignmentCreateParameters(\n",
+    "        role_definition_id=role_def.id,\n",
+    "        principal_id=uai_principal_id,\n",
+    "        principal_type=\"ServicePrincipal\",\n",
+    "    ),\n",
+    ")\n",
+    "print(\"Role assignment for `Storage Blob Data Reader` at the workspace storage account completed.\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 9. Create Safety-Enabled LLaMa Online Endpoint using above score.py"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### 9.1 Decide on SKU and instance count for the Safety-Enabled LLaMa Online Endpoint."
```

Comment:
Similar comment, we should have SKU and instance recommendations. We can loop in training team for these values or look at the model tags.

Reply:
Added